mininet/node.py

```diff
@@ -879,36 +898,42 @@ def setHID( self, *hid ):
             self.cmd( 'xip hid new', p )
             self.cmd( 'xip hid addaddr', p )
 
-    def setLPM( self, **lpm ):
-        """Set routes with Longest Prefix Match
-           Extends CIDR abilities to XIA."""
+    def setLPM( self, *lpm ):
+        """Set routes with Longest Prefix
+           Match. Extends CIDR abilities to XIA."""
         moduleDeps( add='xia_ppal_lpm' )
-        lpmid = lpm.get( 'lpmid' )
-        prefixLen = lpm.get( 'prefixLen' )
-        if not isinstance( prefixLen, basestring ):
-            str( prefixLen )
-        if any( i is None for i in [ lpmid, prefixLen ] ):
-            return
-        self.cmd( 'xip lpm addlocal', lpmid, prefixLen )
+        for l in lpm:
+            if '/' in l:
+                lpmid, prefixLen = l.split( '/' )
+                prefixLen = int( prefixLen )
+            else:
+                error( 'No prefix length provided for lpm: %s' % l )
+                return
+            # Here we do not need to test the length of lpm
+            if self.validate( xid=lpmid, base=16 ) and prefixLen in range( 0, 161 ):
+                self.cmd( 'xip lpm addlocal', lpmid, prefixLen )
+            else:
+                error( 'Not a valid LPM identifier: %s or prefixLen %d \n' % ( lpmid, prefixLen ) )
 
     def setU4ID( self, **u4id ):
         "Create UDP socket for ip:port tuple"
         moduleDeps( add='xia_ppal_u4id' )
-        intf = u4id.get( 'intf' )
+        ipaddr = u4id.get( 'ipaddr' )
         port = u4id.get( 'port' )
         tunnel = u4id.get( 'tunnel', False )
         # UDP checksumming is set by default
         checksum = u4id.get( 'checksum', True )
-        if any( i is None for i in [ intf, port ] ):
+        if any( i is None for i in [ ipaddr, port ] ):
+            error( 'Not enough inputs for U4ID configuration \n' )
             return
         cmd = [ 'xip u4id add' ]
-        if isinstance( intf, basestring ) and '.' in intf:
-            cmd.append( intf )
+        if isinstance( ipaddr, basestring ) and '.' in ipaddr:
+            cmd.append( ipaddr )
         else:
-            intfName = '%s-%s' %( self.__str__(), intf )
-            intf = self.IP( intfName )
-            cmd.append( intf )
-        cmd.append( port )
+            error( 'Not a valid ip for U4ID: %s \n' % ipaddr )
+            return
+        if self.validate( xid=port, base=16, length=6 ):
```

Comment:
I think renaming the utility function from validateXID to validate might just be fine. Thoughts?